src/system/api/rqidApi.ts

```diff
@@ -114,18 +116,65 @@ export class Rqid {
       result.splice(0, 0, ...worldDocuments);
     }
 
+    const distinctWorldRqids: string[] = [
+      ...new Set(result.map((d) => d.data?.flags?.rqg?.documentRqidFlags?.id)),
+    ];
+
     if (["match", "all", "compendiums"].includes(scope)) {
-      const compendiaDocuments = await Rqid.documentsFromCompendia(
-        rqidRegex,
-        rqidDocumentType,
-        lang
-      );
+      let compendiaDocuments = await Rqid.documentsFromCompendia(rqidRegex, rqidDocumentType, lang);
+
       result.splice(result.length, 0, ...compendiaDocuments);
     }
 
     return result;
   }
 
+  /**
+   * Gets only the highest priority documents for each rqid that matches the Regex and
+   * language, with the highest priority documents in the World taking precedence over
+   * any documents
+   * in compendium packs.
+   * @param rqidRegex regex used on the rqid
+   * @param rqidDocumentType the first part of the wanted rqid, for example "i", "a", "je"
+   * @param lang the language to match against ("en", "es", ...)
+   * **match** same logic as fromRqid function,
+   * **all**: find in both world & compendia,
+   * **world**: only search in world,
+   * **compendiums**: only search in compendiums
```

Comment:
this is a description of scope that isn't a parameter of this function.

Reply:
Removed